api/v1alpha1/jobset_webhook.go

```diff
@@ -66,7 +71,20 @@ func (r *JobSet) ValidateCreate() error {
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *JobSet) ValidateUpdate(old runtime.Object) error {
-	return nil
+	oldObjCopy := old.DeepCopyObject().(*JobSet)
+	var allErr []error
+	var updatableFields = []string{
+		"`template.spec.template.spec.nodeSelector`",
+	}
+	for index := range r.Spec.ReplicatedJobs {
```

Comment:
I forgot, we should allow updating nodeSelectors only if the jobSet is suspended.

Reply:
pls add an integration test to the webhook as well to verify that updating nodeSelectors is allowed only when JobSet is suspended